### ISSUE TITLE
Why do `.semijoin()` `.antijoin()` take collection vs arrangement

### ISSUE THREAD

**Issue by ryzhyk** [TimelyDataflow/differential-dataflow#130]
`semijoin()` (and, hence, `antijoin()`, which is implemented in terms of `semijoin`)  takes a collection as its `other` argument, just to instantly apply `arrange_by_self()` to it. I wonder if there is a reason it does not take an arrangement instead in the interests of enabling arrangement sharing?

Thanks!

**Comment by frankmcsherry** [comment]
The short version is that Rust does not support operator overloading, so `semijoin` can have exactly one signature. We could have a `semijoin_arranged`, but for the moment `join_core` fills that role.

**Comment by ryzhyk** [comment]
Oh, I was not suggesting overloading, but rather having two separate versions as you suggest (or only having one version that takes an arrangement). I can of course write my own implementation. But really I was wondering if I am missing something fundamental here.

**Comment by frankmcsherry** [comment]
Nope, nothing fundamental! When I need to do semijoin with arranged data I use `join_core`. It's all an ergonomic work in progress, for which I apologize. :D

**Comment by ryzhyk** [comment]
Thanks!
